src/containers/Network/UpgradeStatus.tsx

```diff
@@ -14,46 +14,87 @@ import {
 } from '../shared/utils'
 import { useLanguage } from '../shared/hooks'
 import Log from '../shared/log'
-import { StreamValidator, ValidatorResponse } from '../shared/vhsTypes'
+import {
+  NodeData,
+  NodeResponse,
+  StreamValidator,
+  ValidatorResponse,
+} from '../shared/vhsTypes'
 import NetworkContext from '../shared/NetworkContext'
+import { ledgerCompare } from './Nodes'
 
 interface DataAggregation {
   label: string
-  value: number
-  count: number
+  validatorsPercent: number
+  validatorsCount: number
+  nodesPercent: number
+  nodesCount: number
 }
 
 export const aggregateData = (
   validators: ValidatorResponse[],
+  nodes: NodeResponse[],
 ): DataAggregation[] => {
   if (!validators) {
     return []
   }
-  let total = 0
-  const aggregation: Record<string, number> = {}
-  validators.forEach((validator) => {
+  let totalVals = 0
+  let totalNodes = 0
+  interface aggregationTypes {
+    validatorsCount: number
+    nodesCount: number
+  }
+
+  const aggregation: Record<string, aggregationTypes> = {}
+  validators?.forEach((validator) => {
     if (validator.signing_key) {
-      total += 1
+      totalVals += 1
       const version = validator.server_version
       if (version) {
         if (!aggregation[version]) {
-          aggregation[version] = 1
+          aggregation[version] = { validatorsCount: 1, nodesCount: 0 }
+        } else {
+          aggregation[version].validatorsCount += 1
+        }
+      }
+    }
+  })
+
+  nodes?.forEach((node) => {
+    if (node.node_public_key) {
+      totalNodes += 1
+      const { version } = node
+      if (version) {
+        if (!aggregation[version]) {
+          aggregation[version] = { validatorsCount: 0, nodesCount: 1 }
```

Comment:
you can simplify by having it create a new object with both 0 and then always increment.